architecture-compose/src/androidLibMain/kotlin/com/splendo/kaluga/architecture/compose/navigation/KalugaNavigatorComposeAdapter.kt

```diff
@@ -0,0 +1,85 @@
+package  com.splendo.kaluga.architecture.compose.navigation
+
+import androidx.compose.runtime.Composable
+import androidx.navigation.NavGraphBuilder
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import com.splendo.kaluga.architecture.navigation.NavigationAction
+import com.splendo.kaluga.architecture.navigation.Navigator
+import kotlin.reflect.KClass
+
+const val BACK_ROUTE = "back"
+private const val EMPTY_ROUTE = "empty"
+
+/** @return a route represented by the [NavigationAction]. */
+inline fun <reified T: NavigationAction<*>> route(vararg arguments: String?): String =
+    route(T::class, * arguments)
+
+/** @return a route represented by this [NavigationAction]. */
+fun NavigationAction<*>.route(vararg arguments: String?): String = route(this::class, *arguments)
+
+/** @return a route represented by [navigationActionClass] and [arguments]. */
+fun route(navigationActionClass: KClass<out NavigationAction<*>>, vararg arguments: String?): String {
+    var route = navigationActionClass.simpleName!!
+    for (argument in arguments) {
+        route += "/$argument"
+    }
+
+    return route
+}
+
+/**
+ * Routes [Navigator] calls to [NavHostController].
+ */
+open class KalugaNavigatorComposeAdapter<A : NavigationAction<*>>(
+    protected val navController: NavHostController,
+    private val navigationMapper: (A) -> String?,
+    private val parentNavigator: KalugaNavigatorComposeAdapter<*>? = null
+) : Navigator<A> {
+
+    override fun navigate(action: A) {
+        val newRoute = navigationMapper(action)!!
+        if (isNewRoute(newRoute)) {
+            if (newRoute == BACK_ROUTE) {
+                back()
+            } else {
+                navigate(newRoute)
+            }
```

Comment:
`NavHostController.navigate` throws `IllegalArgumentException` if route is invalid. What happens when code does `navigate(EMPTY_ROUTE)` and we didn't map it?

Reply:
fixed `navigationMapper` signature to make route non nullable. as for `navigate(newRoute)` - it'd throw `IllegalArgumentException` if the route returned by `navigationMapper` is not present in navigation graph